BritneyS/GetFed
# Core data

## What you did :question:
- Added Core Data functionality so that custom user food entries are saved in Core Data's SQLite database
- Added alerts for both successful and failing saves

## How you did it :white_check_mark:
- Created managedContext object to hold values for Core Data Entities
- Saved the entries for the fields in the Food Entry view to the appropriate entities, Food and Nutrients, and linked the entities (one-to-one relationship); method is called upon tapping 'Save'
- Wrote method to clear textfields upon successful save
- Wrote methods to call alerts for successful and failed saves
- Added temporary method `readRecords()` to display all current records in the data store (to test the save feature)


## How to test it :microscope:
- Run the app
- Tap 'Food Entry'
- Enter values in the Food Entry's fields
- Tap 'Save'
- Note that a success alert message is displayed, and the message includes the name of the food entry
- Note that in the Terminal, the values of the new food entry (starts with 🍞) and the current contents of the data store (starts with 🍎) are displayed. The new food entry should appear at the end of the 🍎list


## Screenshots (if applicable) :camera:
![simulator screen shot - iphone xr - 2018-12-11 at 11 18 09](https://user-images.githubusercontent.com/8409475/49822613-9a6cdb80-fd4b-11e8-957e-125dfeb1841e.png)
![simulator screen shot - iphone xr - 2018-12-11 at 11 18 17](https://user-images.githubusercontent.com/8409475/49822619-a3f64380-fd4b-11e8-9291-9e1615557bdd.png)
![simulator screen shot - iphone xr - 2018-12-11 at 11 18 23](https://user-images.githubusercontent.com/8409475/49822661-be302180-fd4b-11e8-91f1-6e2ab9697983.png)
![screen shot 2018-12-11 at 1 53 00 pm](https://user-images.githubusercontent.com/8409475/49822806-267f0300-fd4c-11e8-967c-22a17186dcd7.png)
